python/torch_mlir/eager_mode/torch_mlir_dispatch.py

```diff
@@ -0,0 +1,406 @@
+import inspect
+import re
+import warnings
+import weakref
+from typing import Optional, List, Dict
+from typing import cast, Tuple, Any, Callable
+
+import numpy as np
+import torch
+import torch._C
+from torch.fx.node import map_aggregate
+from torch.fx.operator_schemas import (
+    _args_kwargs_to_normalized_args_kwargs,
+    type_matches,
+    get_signature_for_torch_op,
+    create_type_hint,
+    ArgsKwargsPair,
+)
+from torch.jit import ScriptFunction
+from torch.utils._pytree import tree_map
+
+from torch_mlir.eager_mode.lazytensor.builder import _get_func_op_with_name
+from torch_mlir import ir
+from torch_mlir.dialects import torch as torch_dialect
+from torch_mlir.dialects.torch.importer.jit_ir import ModuleBuilder
+from torch_mlir.dialects.torch.importer.jit_ir.build_tools.torch_ods_gen import (
+    load_registry,
+)
+from torch_mlir.eager_mode.annotator import Annotation, AnnotationConverter
+from torch_mlir.eager_mode.torch_mlir_types import TorchTensorType
+from torch_mlir_e2e_test.utils import run_pipeline_with_repro_report
+
+SUPPORTED_OPS = frozenset(
+    [
+        member.OPERATION_NAME
+        for member in vars(torch_dialect).values()
+        if hasattr(member, "OPERATION_NAME")
+    ]
+)
+
+
+def check_supported_op(schema: torch._C.FunctionSchema) -> bool:
+    return (
+        "torch."
+        + schema.name.replace("::", ".")
+        + ("." + schema.overload_name if schema.overload_name else "")
+    ) in SUPPORTED_OPS
+
+
+boolean_dispatched: "weakref.WeakKeyDictionary[Callable, Dict[str, Callable]]" = weakref.WeakKeyDictionary()  # noqa: T484
```

Comment:
what is this noqa thing? Is any of this code copied? If so, there are licensing implications we need to sort out, such as by isolating the copied code into a separate file like I did for upstream_shape_functions.py in my PR.